### PR TITLE
PA-2165 Fixed Interest from Enhanced Referral

### PR DESCRIPTION
**Pull request by Fosol**
Interest columns from SPP import were not present in the SPL report.

Updated Project Import to include `Interest from Enhanced Referral`.

image

### CODE REVIEW

File: backend/api/Areas/Project/Mapping/Dispose/ProjectMap.cs [hunk 1]
```diff
@@ -70,6 +70,16 @@ public void Register(TypeAdapterConfig config)
                 .Map(dest => dest.ExemptionRationale, src => src.GetNoteText(Entity.NoteTypes.Exemption))
                 .Map(dest => dest.ReportingNote, src => src.GetNoteText(Entity.NoteTypes.Reporting))
                 .Map(dest => dest.RemovalFromSplRationale, src => src.GetNoteText(Entity.NoteTypes.SplRemoval))
+                .Map(dest => dest.InterestFromEnhancedReferralNote, src => src.GetNoteText(Entity.NoteTypes.AgencyInterest))
+                .Map(dest => dest.DocumentationNote, src => src.GetNoteText(Entity.NoteTypes.Documentation))
```
Comment (Fosol) [inline backend/api/Areas/Project/Mapping/Dispose/ProjectMap.cs:74]: Flattening notes... I hate this approach, but I don't have time to modify the frontend to better handle arrays within Formik.

File: backend/api/Areas/Reports/Models/Project/SurplusPropertyListModel.cs [hunk 2]
```diff
@@ -124,11 +124,11 @@ public class SurplusPropertyListModel
 
         [DisplayName("Interest from Enhanced Referral")]
         [CsvHelper.Configuration.Attributes.Name("Interest from Enhanced Referral")]
-        public string AgencyResponseNote { get; set; }
+        public string InterestFromEnhancedReferralNote { get; set; }
```
Comment (Fosol) [inline backend/api/Areas/Reports/Models/Project/SurplusPropertyListModel.cs:127]: Renamed to better match current use

File: backend/api/Areas/Tools/Helpers/ImportProjectsHelper.cs [hunk 3]
```diff
@@ -241,7 +241,7 @@ private Entity.Project Merge(Entity.Project project, Model.ImportProjectModel mo
                 SixtyDayNotificationSentOn = null, // Don't have a source for this information.
                 NinetyDayNotificationSentOn = null, // Don't have a source for this information.
                 OnHoldNotificationSentOn = null, // Don't have a source for this information.
-                InterestedReceivedOn = null, // Don't have a source for this information.
+                InterestedReceivedOn = model.InterestedReceivedOn,
```
Comment (Fosol) [inline backend/api/Areas/Tools/Helpers/ImportProjectsHelper.cs:244]: Import the value from the SIS data

File: frontend/src/features/projects/common/forms/DocumentationForm.tsx [hunk 4]
```diff
@@ -18,13 +17,7 @@ const DocumentationForm = ({ isReadOnly, tasks, showNote = false }: IDocumentati
     <Fragment>
       <h3>Documentation</h3>
       <TasksForm tasks={tasks ?? []} isReadOnly={isReadOnly} />
-      {showNote && (
-        <ProjectNotes
-          label="Note"
-          field={`notes[${NoteTypes.Documentation}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/common/forms/DocumentationForm.tsx:24]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx [hunk 5]
```diff
@@ -48,6 +49,28 @@ const EnhancedReferralCompleteForm = ({
   return (
     <Container fluid className="EnhancedReferralCompleteForm">
       <h3>Enhanced Referral Process Complete</h3>
+      <Form.Row>
+        <Form.Label column md={4}>
+          Interest Received On
```
Comment (Fosol) [inline frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:54]: Adding interest properties to ERP form

File: frontend/src/features/projects/spl/forms/CloseOutAdjustmentForm.tsx [hunk 6]
```diff
@@ -50,7 +49,7 @@ const CloseOutAdjustment = (props: CloseOutAdjustmentProps) => {
       </Form.Row>
       <ProjectNotes
         label="Adjustment to Prior Year Sale Notes"
-        field={`notes[${NoteTypes.Adjustment}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/forms/CloseOutAdjustmentForm.tsx:53]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/spl/forms/CloseOutFinancialSummaryForm.tsx [hunk 7]
```diff
@@ -124,21 +123,21 @@ const CloseOutFinancialSummaryForm = (props: CloseOutFinancialSummaryFormProps)
         <Col md={1}>&nbsp;</Col>
         <Col>
           <ProjectNotes
-            field={`notes[${NoteTypes.Remediation}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/forms/CloseOutFinancialSummaryForm.tsx:127]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/spl/forms/CloseOutFinancialSummaryForm.tsx [hunk 8]
```diff
@@ -124,21 +123,21 @@ const CloseOutFinancialSummaryForm = (props: CloseOutFinancialSummaryFormProps)
...
+            field="remediationNote"
             label="Remediation (Not funded throuch COS. TRAN only)"
             disabled={props.isReadOnly}
             outerClassName="col"
             className="col-md-10"
           />
           <ProjectNotes
-            field={`notes[${NoteTypes.SplCost}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/forms/CloseOutFinancialSummaryForm.tsx:134]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/spl/forms/CloseOutFinancialSummaryForm.tsx [hunk 9]
```diff
@@ -124,21 +123,21 @@ const CloseOutFinancialSummaryForm = (props: CloseOutFinancialSummaryFormProps)
...
+            field="programCostNote"
             label="SPL Cost Calculation Notes"
             disabled={props.isReadOnly}
             outerClassName="col"
             className="col-md-10"
           />
           <ProjectNotes
-            field={`notes[${NoteTypes.SplGain}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/forms/CloseOutFinancialSummaryForm.tsx:141]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx [hunk 10]
```diff
@@ -16,7 +15,8 @@ const CloseOutFinancialsForm = (props: CloseOutFinancialsFormProps) => {
       <Form.Row>
         <Col>
           <ProjectNotes
-            field={`notes[${NoteTypes.LoanTerms}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx:19]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx [hunk 11]
```diff
@@ -26,7 +26,8 @@ const CloseOutFinancialsForm = (props: CloseOutFinancialsFormProps) => {
         <Col md={1}></Col>
         <Col>
           <ProjectNotes
-            field={`notes[${NoteTypes.CloseOut}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx:29]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/spl/tabs/CloseOutFormTab.tsx [hunk 12]
```diff
@@ -37,7 +36,7 @@ const CloseOutFormTab: React.FunctionComponent<ICloseOutFormTabProps> = ({
       <Form.Row>
         <Col>
           <ProjectNotes
-            field={`notes[${NoteTypes.SalesHistory}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/tabs/CloseOutFormTab.tsx:40]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/spl/tabs/CloseOutFormTab.tsx [hunk 13]
```diff
@@ -47,7 +46,7 @@ const CloseOutFormTab: React.FunctionComponent<ICloseOutFormTabProps> = ({
         <Col md={1}></Col>
         <Col>
           <ProjectNotes
-            field={`notes[${NoteTypes.Comments}].note`}
```
Comment (Fosol) [inline frontend/src/features/projects/spl/tabs/CloseOutFormTab.tsx:50]: Using array positioning is an invalid implementation and will result in data corruption

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx [hunk 14]
```diff
@@ -48,6 +49,28 @@ const EnhancedReferralCompleteForm = ({
...
+          field="interestedReceivedOn"
+        />
+      </Form.Row>
+      <Form.Row>
+        <Col>
+          <ProjectNotes
+            field="interestFromEnhancedReferralNote"
+            label="Interest Note"
+            disabled={isReadOnly}
+            outerClassName="col"
```
Comment (devinleighsmith) [inline frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:69]: is col just making this a 100% width field?

File: frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx [hunk 15]
```diff
@@ -16,7 +15,8 @@ const CloseOutFinancialsForm = (props: CloseOutFinancialsFormProps) => {
       <Form.Row>
         <Col>
           <ProjectNotes
-            field={`notes[${NoteTypes.LoanTerms}].note`}
+            data-testid="loanTermsNote"
+            field="loanTermsNote"
             label="Loan Terms"
```
Comment (devinleighsmith) [inline frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx:20]: Ideally, you should be able to get the note field via this label. If that isn't working we should try and fix the label.

**Review by devinleighsmith — APPROVED** [review]

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx (hunk 14 above)
Comment (Fosol) [reply frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:69]: without it the text area doesn't align with the other inputs on the form.

File: frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx (hunk 15 above)
Comment (Fosol) [reply frontend/src/features/projects/spl/forms/CloseOutFinancialsForm.tsx:20]: After your comment about how the testing library wants to work I instead looked for the label.  It solved the issue.

**Pull request merged**